### PR TITLE
Update NUTS3 shapes

### PR DESCRIPTION
**Pull request by cpschau**
Closes #1242

## Changes proposed in this Pull Request

The NUTS3 and NUTS2 shapes have been updated from a 1:60M resolution (red) to a more detailed 1:3M resolution (blue). The data is now directly retrieved from GISCO, making the previously used resources, which are part of the data bundle hosted on Zenodo, redundant and removable.

Screenshot from 2024-08-28 16-31-20
Screenshot from 2024-08-28 16-29-26
Screenshot from 2024-08-28 16-28-40

## Checklist

- [x] I tested my contribution locally and it works as intended.
- [x] Code and workflow changes are sufficiently documented.
- [x] Changed dependencies are added to `envs/environment.yaml`.
- [x] Changes in configuration options are added in `config/config.default.yaml`.
- [x] Changes in configuration options are documented in `doc/configtables/*.csv`.
- [x] Sources of newly added data are documented in `doc/data_sources.rst`.
- [x] A release note `doc/release_notes.rst` is added.

### CODE REVIEW

File: rules/retrieve.smk [hunk 1]
```diff
@@ -77,6 +77,32 @@ if config["enable"]["retrieve"] and config["enable"].get("retrieve_databundle",
             "../scripts/retrieve_eurostat_household_data.py"
 
 
+if config["enable"]["retrieve"]:
+
+    rule retrieve_shapes_data:
```
Comment (fneum) [inline rules/retrieve.smk:82]:
```suggestion
    rule retrieve_nuts_shapes:
```

**Review by fneum — COMMENTED** [review]
Is "nuts/NUTS_RG_10M_2013_4326_LEVL_2.geojson" (which is currently in the data bundle) actually also included in the downloaded ZIP? Then one could include it here as well and remove from the data bundle.

Then they could both be extracted into `data/nuts`

**Comment by cpschau** [comment]
> Is "nuts/NUTS_RG_10M_2013_4326_LEVL_2.geojson" (which is currently in the data bundle) actually also included in the downloaded ZIP? Then one could include it here as well and remove from the data bundle.
> 
> Then they could both be extracted into `data/nuts`

The NUTS_RG_03M_2013_4326_LEVL_2.geojson with (also with 1:3M not 1:10M) is included and could be co-retrieved.

**Review by fneum — APPROVED** [review]
Excellent!

**Pull request merged**